MergerService/Utils/TaskUtils.cs

```diff
@@ -1,46 +1,172 @@
+                // activity.AddTag("progress", progress);
+
+                using (var content = new StringContent(JsonConvert.SerializeObject(new
+                {
+                    percentage = progress
+                }, this._jsonSerializerSettings)))
+                {
+                    content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                    Update(jobId, taskId, content);
+                }
+            }
+        }
+
+        public void UpdateCompletion(string jobId, string taskId)
+        {
+            using (this._activitySource.StartActivity("update task completed"))
+            {
+                using (var content = new StringContent(JsonConvert.SerializeObject(new
+                {
+                    percentage = 100,
+                    status = Status.COMPLETED
+                }, this._jsonSerializerSettings)))
+                {
+                    content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                    Update(jobId, taskId, content);
+                }
+            }
+        }
+
+        public void UpdateReject(string jobId, string taskId, int attempts, string reason, bool resettable)
+        {
+            using (var activity = this._activitySource.StartActivity("reject task"))
+            {
+                // activity.AddTag("attempts", attempts);
+                // activity.AddTag("resettable", resettable);
+
+                // Check if the task should actually fail
+                if (!resettable || attempts == this._maxAttempts)
+                {
+                    UpdateFailed(jobId, taskId, attempts, reason, resettable);
+                    return;
+                }
+
+                attempts++;
+
+                using (var content = new StringContent(JsonConvert.SerializeObject(new
+                {
+                    status = Status.PENDING,
+                    attempts,
+                    reason,
+                    resettable
+                }, this._jsonSerializerSettings)))
+                {
+                    content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                    Update(jobId, taskId, content);
+                }
+            }
+        }
+
+        private void UpdateFailed(string jobId, string taskId, int attempts, string reason, bool resettable)
+        {
+            using (var activity = this._activitySource.StartActivity("fail task"))
+            {
+                // activity.AddTag("reason", reason);
+
+                attempts++;
+
+                using (var content = new StringContent(JsonConvert.SerializeObject(new
+                {
+                    status = Status.FAILED,
+                    attempts,
+                    reason,
+                    resettable
+                }, this._jsonSerializerSettings)))
+                {
+                    content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
```

Comment:
do you have a static or consts that "application/json" can move to?

Reply:
No, and i don't think there is a reason to.